### ISSUE TITLE
Monthly Milestone - November 2019

### ISSUE THREAD

**Issue by emonddr** [strongloop/loopback-next#4029]
# November Milestone

## Goals

_Inclusion of related models MVP [#1352]_
- [x] Run repository tests for Cloudant #3437 
- [x] Verify relation type in `resolve{Relation}Metadata` #3440 
- [ ] 🏃Reject create/update requests when data contains navigational properties #3439 

_Migration guide between LB3 and LB4 MVP [#453 ]_ 

- [ ]  🏃How to migrate (app-level) Express middleware from LB3 to LB4 #3947 
- [x] How to migrate model relations #3948 

_Authentication & Authorization [#3902 ]_ 

- [ ] 🏃 Update the authorization tutorial, https://github.com/strongloop/loopback-next/issues/3694
- [x] Apply authorization in the shopping example #3695 
- [x]  🏃Refactoring: Customer credentials #1996 

_Add Support for Partitioned Database_ [#219](https://github.com/strongloop/loopback-connector-cloudant/issues/219)

- [x] Support creating partitioned index [#216](https://github.com/strongloop/loopback-connector-cloudant/issues/216)
- [x] findAll: Read partitionKey from `options` [#220](https://github.com/strongloop/loopback-connector-cloudant/issues/220) blocked by #216

_From model definition to REST API with no custom repository/controller classes [#2036 ]_

- [ ]  🏃Model API booter & builder #3736 

_Bugs_

- [x] postgres connector not returning generated identifier value on create of entity for camelCased identifiers in table #3749 
- [x]  🏃String type in the generated model class from lb4 discover not correct #3806 
- [x] @param.path generated with lb4 relation considers Wrapper datatypes #3711  (wait until #3806 is finished)

_Miscellaneous_ 

- [x] Update loopback-component-explorer README [#432](https://github.com/strongloop-internal/scrum-apex/issues/432)
- [x] 🏃[Spike] Investigate how to allow complex object in filter object in API Explorer #3770 
- [x] Docs: Update TodoList example and model relations pages to use `lb4 relation` command #3742 
- [x] Add loopback-connector-openapi and loopback-connector-grpc to be part of the available connectors https://github.com/strongloop/loopback-next/issues/3714
- [ ] Create Skills Transfer slides for support team https://github.com/strongloop-internal/scrum-apex/issues/437

## Stretch Goals 

_Add Support for Partitioned Database_ [#219]
(https://github.com/strongloop/loopback-connector-cloudant/issues/219)

- [x] Optimize query by reading partition key from payload  [#217](https://github.com/strongloop/loopback-connector-cloudant/issues/217)

_From model definition to REST API with no custom repository/controller classes [#2036]_

- [ ] Add CrudRestApiBuilder to `@loopback/rest-crud`  #3737 

_Authentication & Authorization [#3902 ]_ 

- [ ] Contribute OpenAPI spec pieces from extensions #3854 
- [ ] Add user profile factory for authentication modules #3846 

_Production Deployment_ [#1054](https://github.com/strongloop/loopback-next/issues/1054)

- [x] [Blog post] Show how to deploy a LB application in a cloud native environment #3715

**Comment by dhmlau** [comment]
Moved all the outstanding items to Dec.  Closing this as November ended. Thanks team!
